Tella/Scenes/Uwazi/ViewModel/SubmittedEntityViewModel.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  SubmittedEntityViewModel.swift
+//  Tella
+//
+//  Created by Dhekra Rouatbi on 10/5/2024.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class SubmittedEntityViewModel: ObservableObject {
+    
+    var mainAppModel : MainAppModel
+    var entityInstance: UwaziEntityInstance?
+    var uwaziSubmissionViewModel: UwaziSubmissionViewModel?
+    private var subscribers = Set<AnyCancellable>()
+    var uwaziVaultFiles : [UwaziVaultFile] = []
+    @Published var shouldShowToast : Bool = false
+    @Published var toastMessage : String = ""
+
+    var entityTitle : String {
+        guard let stringValue = self.entityInstance?.title else { return "" }
+        return stringValue
+    }
+    
+    var templateName : String {
+        String(format: "%@ %@", LocalizableUwazi.uwaziEntitySummaryDetailTemplateTitle.localized, entityInstance?.collectedTemplate?.entityRow?.name ?? "")
+    }
+    
+    var uploadedOn : String {
+        guard let updatedDate = entityInstance?.updatedDate else { return "" }
+        return String(format: LocalizableUwazi.uploadedOn.localized, updatedDate.getFormattedDateString(format: DateFormat.submittedReport.rawValue))
+    }
+    
+    var filesDetails : String {
+        if uwaziVaultFiles.count > 1 {
+            let totalSize = self.uwaziVaultFiles.reduce(0) { $0 + $1.size}
+            let fileString = uwaziVaultFiles.count > 1 ? LocalizableUwazi.submittedFiles.localized : LocalizableUwazi.submittedFile.localized
+            return String(format: fileString,  uwaziVaultFiles.count, totalSize.getFormattedFileSize() )
+        }  else {
+            return ""
+        }
+    }
+    
+    var tellaData: TellaData? {
+        return self.mainAppModel.tellaData
+    }
+    
+    init(mainAppModel : MainAppModel,
+         entityInstance: UwaziEntityInstance? = nil,
+         entityInstanceId: Int? = nil) {
+        self.mainAppModel = mainAppModel
+        
+        if let entityInstance {
+            self.entityInstance = entityInstance
```

Comment:
In the initializer, if entityInstance is provided, you can directly set it and return to avoid unnecessary checks ...

Reply:
I think we need to affect the `entityInstance` from the initialiser `if` it's different to nil `else` get the instance from database ?